openwisp_users/utils.py

```diff
@@ -57,3 +59,12 @@ def usermodel_list_and_search(model, additional_fields):
         displays = model.list_display[:]
         model.list_display = displays[: field[0]] + [field[1]] + displays[field[0] :]
         model.search_fields += (field[1],)
+
+
+def validate_user_groups(data):
```

Comment:
please move this to `admin.py`, we can use a mixin.

`EmailRequiredMixin` is currently being inherited by both forms, so we could do change that to:

```python
class UserFormMixin(forms.ModelForm):
    email = forms.EmailField(label=_('Email'), max_length=254, required=True)

    def validate_user_groups(data):
        is_staff = data.get('is_staff')
        groups = data.get('groups')
        if is_staff and not groups:
            raise ValidationError(
                {'groups': _('A staff must belong to a group, Please select one')}
            )

     def clean(self):
        cleaned_data = super().clean()
        validate_user_groups(cleaned_data)
        return cleaned_data
```